pkgs/applications/networking/cluster/tfctl/default.nix

```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+}:
+buildGoModule rec {
+  name = "tfctl";
+  version = "0.15.1";
+  src = fetchFromGitHub {
+    owner = "flux-iac";
+    repo = "tofu-controller";
+    rev = "v${version}";
+    sha256 = "rA3HLO4sD8UmcebGGFxyg0zFpDHCzFHjHwMxPw8MiRU=";
+  };
+  vendorHash = "sha256-NhXgWuxSuurP46DBWOviFzCINJKaTb1mINRYeYcnnH8=";
+
+  subPackages = ["cmd/tfctl"];
+  ldflags = [
+    "-X main.BuildSHA=${src.rev}"
+    "-X main.BuildVersion=${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    for shell in bash zsh fish ; do
+      $out/bin/tfctl completion "$shell" > "tfctl.$shell"
+    done
+
+    installShellCompletion tfctl.{bash,zsh,fish}
+  '';
+
+  meta = with lib; {
```

Comment:
Lets avoid `with lib;` around the meta block, as it is a old and discouraged way, which has shown some issues over the years.